platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -57,3 +57,57 @@ include::sample[dir="snippets/bestPractices/usePluginsBlock/groovy/do-this",file
 
 === Tags
 `#structuring-builds`
+
+[[use_convention_plugins]]
+== Use Convention Plugins for Common Build Logic
+
+You should use convention plugins to encapsulate and reuse common build logic across multiple build files.
+
+=== Explanation
+
+Convention plugins help you avoid duplicating build logic across multiple build files by moving shared configuration into a reusable plugin. This approach:
+
+* Reduces duplication and makes build scripts more maintainable
+* Centralizes common configuration, making it easier to update
+* Makes build files cleaner and more focused on project-specific configuration
+* Enables better IDE support and validation for your build logic
+
+Although the initial setup of a convention plugin requires some effort, it significantly simplifies maintenance and addition of new build logic in the long run.
+
+=== Example
+
+==== Don't Do This
+
+====
+include::sample[dir="snippets/bestPractices/useConventionPlugins/kotlin/avoid-this",files="build.gradle.kts[tags=avoid-this]"]
+include::sample[dir="snippets/bestPractices/useConventionPlugins/groovy/avoid-this",files="build.gradle[tags=avoid-this]"]
+====
+
+<1> Common compiler settings duplicated across multiple build files
+<2> Test configuration that needs to be maintained in multiple places
+<3> Common dependencies that could be centralized
+
+==== Do This Instead
+
+Create a convention plugin in included build:
```

Comment:
```suggestion
Create a convention plugin in `buildSrc`:
```

Reply:
https://github.com/gradle/gradle/pull/33081 What is gradle’s official stance on this? The messaging seems muddled especially after Herding Elephants

Reply:
These 2 items should be aligned.  I think this item should use a `build-logic` included build for the convention plugins.

Reply:
Ok, I'll rework this and update the message.

Reply:
Or if decision on #33081 is on hold, I think it is logical to wait here for resolution as well, isn't it?

Reply:
I've decided to switch to build-logic already.  We may keep this best practice on hold until we resolve #33081 . Switching back to buildSrc shouldn't be a problem.

Reply:
Yeah, sorry we can't give any clarity about this yet.